controllers/addresspool_controller.go

```diff
@@ -62,20 +62,35 @@ func (r *AddressPoolReconciler) UpdateInsyncStatus(client *gophercloud.ServiceCl
 // Reconciled/Insync to false only if Generation != ObservedGeneration.
 func (r *AddressPoolReconciler) ReconcileResource(client *gophercloud.ServiceClient, instance *starlingxv1.AddressPool, request_namespace string) (err error) {
 	if instance.DeletionTimestamp.IsZero() {
-		host_instance := &starlingxv1.Host{}
-		host_namespace := types.NamespacedName{Namespace: instance.Namespace, Name: "controller-0"}
-		err := r.Client.Get(context.TODO(), host_namespace, host_instance)
-		if err != nil {
-			logAddressPool.Error(err, "Failed to get host resource from namespace")
-		}
+		if instance.Status.ObservedGeneration != instance.ObjectMeta.Generation {
+			host_instance, err := r.CloudManager.GetActiveHost(request_namespace, client)
+			if err != nil {
+				msg := "failed to get active host"
+				return common.NewUserDataError(msg)
+			}
 
-		host_instance.Status.InSync = false
-		host_instance.Status.Reconciled = false
-		err = r.Client.Status().Update(context.TODO(), host_instance)
-		if err != nil {
-			logAddressPool.Error(err, "Failed to update status")
-			return err
+			host_instance.Status.InSync = false
+			host_instance.Status.Reconciled = false
+			err = r.Client.Status().Update(context.TODO(), host_instance)
+			if err != nil {
+				msg := fmt.Sprintf("Failed to update '%s' host instance status", host_instance.Name)
+				logAddressPool.Error(err, msg)
+				return common.NewResourceConfigurationDependency(msg)
+			} else {
```

Comment:
dont need the else (return inside the if), please avoid to turn the methods more nested.

Reply:
Done